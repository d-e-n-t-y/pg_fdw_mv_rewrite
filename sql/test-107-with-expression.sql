explain (VERBOSE, COSTS OFF) select key, case when count(value) < 10 then 'several'
        when count(value) < 100 then 'tens'
        when count(value) < 1000 then 'hundreds'
        when count(value) < 10000 then 'thousands'
        when count(value) < 100000 then 'tens of thousands'
        when count(value) < 1000000 then 'hundreds of thousands'
        when count(value) < 10000000 then 'millions'
        else 'many'
    end as quantity from test group by key;

select key, case when count(value) < 10 then 'several'
        when count(value) < 100 then 'tens'
        when count(value) < 1000 then 'hundreds'
        when count(value) < 10000 then 'thousands'
        when count(value) < 100000 then 'tens of thousands'
        when count(value) < 1000000 then 'hundreds of thousands'
        when count(value) < 10000000 then 'millions'
        else 'many'
    end as quantity from test group by key;
