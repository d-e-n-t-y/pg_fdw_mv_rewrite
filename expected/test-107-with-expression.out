explain (VERBOSE, COSTS OFF) select key, case when count(value) < 10 then 'several'
        when count(value) < 100 then 'tens'
        when count(value) < 1000 then 'hundreds'
        when count(value) < 10000 then 'thousands'
        when count(value) < 100000 then 'tens of thousands'
        when count(value) < 1000000 then 'hundreds of thousands'
        when count(value) < 10000000 then 'millions'
        else 'many'
    end as quantity from test group by key;
                                                                                                                                                                                                         QUERY PLAN                                                                                                                                                                                                          
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Custom Scan (MVRewriteScan)
   Output: key, (CASE WHEN (count(value) < 10) THEN 'several'::text WHEN (count(value) < 100) THEN 'tens'::text WHEN (count(value) < 1000) THEN 'hundreds'::text WHEN (count(value) < 10000) THEN 'thousands'::text WHEN (count(value) < 100000) THEN 'tens of thousands'::text WHEN (count(value) < 1000000) THEN 'hundreds of thousands'::text WHEN (count(value) < 10000000) THEN 'millions'::text ELSE 'many'::text END)
   Rewritten: scan of public.test_expr_mv1
(3 rows)

select key, case when count(value) < 10 then 'several'
        when count(value) < 100 then 'tens'
        when count(value) < 1000 then 'hundreds'
        when count(value) < 10000 then 'thousands'
        when count(value) < 100000 then 'tens of thousands'
        when count(value) < 1000000 then 'hundreds of thousands'
        when count(value) < 10000000 then 'millions'
        else 'many'
    end as quantity from test group by key;
 key  | quantity 
------+----------
 key1 | tens
 key2 | millions
(2 rows)

